classes/rest/class-rest-api.php

```diff
@@ -141,6 +142,74 @@ public static function endpoints(): void {
 				],
 			]
 		);
+		/**
+		 * Endpoint to retrieve the latest posts for the Articles block
+		 */
```

Comment:
Maybe to make it easier to review, you could rename the function in `class-articles.php` and then call it here instead of moving the code to here? That way it will be a lot easier to see what actually changed about the code.